### PR TITLE
[Core] Sparse multiplication utility using signed integers primarly

### PR DESCRIPTION
**Pull request by loumalouomega**
This cleans the combination of signed and unsigned integers in the construction of sparse matrices (which can be dangerous)

### CODE REVIEW

**Review by RiccardoRossi — COMMENTED** [review]
why are u making this change?

**Comment by loumalouomega** [comment]
Because mixing signed and unsigned can be problematic, and the only reason I was using signed was for some error in the moment

File: kratos/utilities/sparse_matrix_multiplication_utility.h [hunk 1]
```diff
@@ -886,81 +909,94 @@ class SparseMatrixMultiplicationUtility
...
 
-                std::swap(tmp_col1, tmp_col3);
+                std::swap(Tmp1Column, Tmp3Column);
             }
         }
 
         // Merge the tail.
-        a2 = *acol;
-        return MergeRows<false>( tmp_col1, tmp_col1 + c_col1, bcol + bptr[a2], bcol + bptr[a2+1], tmp_col2 ) - tmp_col2;
+        a2 = *AColumn;
```
Comment (RiccardoRossi) [inline kratos/utilities/sparse_matrix_multiplication_utility.h:933]: as usually...by changing the name to the variables you changed something that was relatively easy to review into something that is very hard to follow.

PLEASE avoid doing it!!!

Comment (loumalouomega) [reply kratos/utilities/sparse_matrix_multiplication_utility.h:933]: Sorry, you are right

**Review by RiccardoRossi — APPROVED** [review]

**Comment by loumalouomega** [comment]
Thanks
